### PR TITLE
Allow configuring auto compaction count

### PR DESCRIPTION
**Pull request by vkamra**
This introduces an environment variable to configure manifest auto compaction.

Once this approach is validated - we can follow up with a PR upstream with the environment variable or by plumbing this via repository connect options.

### CODE REVIEW

File: repo/manifest/manifest_manager_test.go [hunk 1]
```diff
@@ -381,3 +382,63 @@ func TestManifestAutoCompaction(t *testing.T) {
...
+
+	// Should not trigger compaction
+	_, err := mgr.Find(ctx, labels1)
+	require.NoError(t, err)
+
+	foundContents := getManifestContentCount(ctx, t, mgr)
+
+	if got, want := foundContents, compactionCount-1; got != want {
+		t.Errorf("unexpected number of blocks: %v, want %v", got, want)
+	}
```
Comment (ashmrtn) [inline repo/manifest/manifest_manager_test.go:412]: can also use an assert here
```suggestion
	assert.Equal(t, compactionCount-1, foundContents, "unexpected number of manifest blobs")
```

File: repo/manifest/manifest_manager_test.go [hunk 2]
```diff
@@ -381,3 +382,63 @@ func TestManifestAutoCompaction(t *testing.T) {
...
+	foundContents := 0
+	if err := mgr.b.IterateContents(
+		ctx,
+		content.IterateOptions{Range: index.PrefixRange(ContentPrefix)},
+		func(ci content.Info) error {
+			foundContents++
+			return nil
+		}); err != nil {
+		t.Errorf("unable to list manifest content: %v", err)
+	}
```
Comment (ashmrtn) [inline repo/manifest/manifest_manager_test.go:442]:
```suggestion
	err := mgr.b.IterateContents(
		ctx,
		content.IterateOptions{Range: index.PrefixRange(ContentPrefix)},
		func(ci content.Info) error {
			foundContents++
			return nil
		})
	require.NoError(t, err "listing manifest content")
```

**Review by ashmrtn — COMMENTED** [review]
just minor test nits about using testify

**Review by ashmrtn — APPROVED** [review]

File: repo/manifest/manifest_manager.go [hunk 3]
```diff
@@ -32,10 +34,23 @@ var ErrNotFound = errors.New("not found")
...
 // ContentPrefix is the prefix of the content id for manifests.
 const (
-	ContentPrefix              = "m"
-	autoCompactionContentCount = 16
+	ContentPrefix                     = "m"
+	autoCompactionContentCountDefault = 16
+	autoCompactionEnvKey              = "KOPIA_MANIFEST_AUTOCOMPACT_COUNT"
 )
 
+// getAutoCompactionContentCount returns the content count after which
```
Comment (jkowalski) [inline repo/manifest/manifest_manager.go:42]: btw, it may actualy be easier to pass the threshold as a parameter to New().

**Pull request closed without merging**